inception/inception-api-annotation-experimental/src/main/ts/client/AnnotationExperienceAPIImpl.ts

```diff
@@ -16,293 +16,448 @@
-            annotationType: aAnnotationType
-        };
+    requestNewDocumentFromServer(aClientName: string, aUserName: string, aProjectId: number, aDocumentId: number, aViewport: Viewport) {
+        const that = this;
+        this._viewport = aViewport;
+        that._stompClient.publish({
+            destination: "/app/new_document_from_client", body: JSON.stringify(
+                new NewDocumentRequest(aClientName, aUserName, aProjectId, aDocumentId, aViewport))
+        });
+    }
 
-        //TODO correct edit
-        //this.stompClient.publish({destination: "/app/select_annotation_by_client", body: JSON.stringify(json)});
+    requestNewViewportFromServer(aClientName: string, aUserName: string, aProjectId: number, aDocumentId: number, aViewport: Viewport) {
+        this._viewport = aViewport;
+        this._stompClient.publish({
+            destination: "/app/new_viewport_from_client",
+            body: JSON.stringify(new NewViewportRequest(aClientName, aUserName, aProjectId, aDocumentId, aViewport))
+        });
     }
 
-    sendDocumentMessageToServer(aUsername, aDocument, aOffset, aOffsetType) {
 
-        let json = {
-            clientName: this.client,
-            username: aUsername,
-            project: this.projectID,
-            document: aDocument,
-            viewport: aOffset,
-            offsetType: aOffsetType
-        }
+    requestSelectSpanFromServer(aClientName: string, aUserName: string, aProjectId: number, aDocumentId: number, aAnnotationAddress: number)
+    {
+        this._stompClient.publish({
+            destination: "/app/select_annotation_from_client",
+            body: JSON.stringify(new SelectSpanRequest(aClientName, aUserName, aProjectId, aDocumentId, aAnnotationAddress))
+        });
+    }
 
-        this.viewport = json.viewport;
-        this.stompClient.publish({destination: "/app/new_document_by_client", body: JSON.stringify(json)});
+    requestUpdateSpanFromServer(aClientName: string, aUserName: string, aProjectId: number, aDocumentId: number, aAnnotationAddress: number, aNewFeature: string[])
+    {
+        this._stompClient.publish({
+            destination: "/app/update_annotation_from_client",
+            body: JSON.stringify(new UpdateSpanRequest(aClientName, aUserName, aProjectId, aDocumentId, aAnnotationAddress, aNewFeature))
+        });
     }
 
-    sendViewportMessageToServer(aUsername, aViewport, aOffsetType) {
-        let json = {
-            clientName: this.client,
-            username: aUsername,
-            project: this.projectID,
-            document: this.documentID,
-            viewport: aViewport,
-            offsetType: aOffsetType
-        };
-        this.viewport = aViewport;
-        this.stompClient.publish({destination: "/app/new_viewport_by_client", body: JSON.stringify(json)});
+    requestCreateSpanFromServer(aClientName: string, aUserName: string, aProjectId: number, aDocumentId: number, aBegin: number, aEnd: number, aLayer: string) {
```

Comment:
Can you react to the creation of the span in any way? E.g. can you request the creation of a span and then when the span has been created follow up with setting features? Would it be sensible to return a promise to which you can react once the server has indicated that the promise has been fulfilled?

Reply:
Yes, as all methods marked with "on.." (e.g. onSpanCreate) are the listeners to any changes (also made by other users) to my current visible document. So the workflow is the following:
------------
I want to create a span 
-> createSpanRequest() 
-> Server receives requests and creates the span 
-> Event on serverside (SpanCreatedEvent) 
-> method onSpanCreatedEventHandler() listens to that event and disseminates the "new span" to all listeners on a specific topic (the client who has created the span is listening to it ofc)
-> Client receives it via onSpanCreate() method
---------------
If the client now wants to set up features, he can simply call the "requestUpdateSpan()" with the new features and the same streamlined process begins for the update